### PR TITLE
Add yhaliaw keymap for Planck/Rev6.1.

### PR DESCRIPTION
**Pull request by yhaliaw**
## Description

Adding a user keymap for Planck Rev6.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* None

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/planck/keymaps/yhaliaw/config.h [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#pragma once
```
Comment (drashna) [inline keyboards/planck/keymaps/yhaliaw/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/planck/keymaps/yhaliaw/keymap.c [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    if (!is_audio_on()) {
+        audio_disable = true;
+        audio_on();
+    }
+#endif
+}
+
+void matrix_init_user(void) {
+    // To Fix a audio related problem. See comments in keyboard_pre_init_user().
+#ifdef AUDIO_ENABLE
```
Comment (drashna) [inline keyboards/planck/keymaps/yhaliaw/keymap.c:148]: That's a nice hack!

File: keyboards/planck/keymaps/yhaliaw/keymap.c [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+}
+
+void keyboard_post_init_user(void) {
+    rgblight_layers = rgb_layers;
+    rgblight_set_layer_state(0, true);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t* record) {
+    switch (keycode) {
+        case CLEAR:
```
Comment (drashna) [inline keyboards/planck/keymaps/yhaliaw/keymap.c:162]: This is actually a keycode already.  `EEP_RST`, specifically.

File: keyboards/planck/keymaps/yhaliaw/keymap.c [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+    // Turn on RBG for capslock.
+    rgblight_set_layer_state(4, led_state.caps_lock);
+    return true;
+}
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    // Set RBG layer according to active keymap layer.
+    rgblight_set_layer_state(1, layer_state_cmp(state, 1));
+    rgblight_set_layer_state(2, layer_state_cmp(state, 2));
+    rgblight_set_layer_state(3, layer_state_cmp(state, 1) && layer_state_cmp(state, 2));
```
Comment (drashna) [inline keyboards/planck/keymaps/yhaliaw/keymap.c:221]: The layers shouldn't be negative, as technically, it's asking for an unsigned integer.

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/planck/keymaps/yhaliaw/config.h (hunk 1 above)
Comment (yhaliaw) [reply keyboards/planck/keymaps/yhaliaw/config.h:17]: I have added the license header in the config.h file.
Thanks for the reminder.

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thank you for your contribution!

**Pull request merged**